pkg/services/authn/grpcutils/config.go

```diff
@@ -0,0 +1,68 @@
+package grpcutils
+
+import (
+	"fmt"
+
+	"github.com/grafana/grafana/pkg/setting"
+)
+
+type Mode string
+
+func (s Mode) IsValid() bool {
+	switch s {
+	case ModeOnPrem, ModeCloud:
+		return true
+	}
+	return false
+}
+
+const (
+	ModeOnPrem Mode = "on-prem"
+	ModeCloud  Mode = "cloud"
+)
+
+type GrpcClientConfig struct {
+	Token            string
+	TokenExchangeURL string
+	TokenNamespace   string
+	Mode             Mode
+}
+
+func ReadGrpcClientConfig(cfg *setting.Cfg) (*GrpcClientConfig, error) {
+	section := cfg.SectionWithEnvOverrides("grpc_client_authentication")
+
+	mode := Mode(section.Key("mode").MustString(string(ModeOnPrem)))
+	if !mode.IsValid() {
+		return nil, fmt.Errorf("grpc_client_authentication: invalid mode %q", mode)
+	}
+
+	return &GrpcClientConfig{
+		Token:            section.Key("token").MustString(""),
+		TokenExchangeURL: section.Key("token_exchange_url").MustString(""),
+		TokenNamespace:   section.Key("token_namespace").MustString("stacks-" + cfg.StackID),
+		Mode:             mode,
+	}, nil
+}
+
+type GrpcServerConfig struct {
+	SigningKeysURL   string
+	AllowedAudiences []string
+	Mode             Mode
+	LegacyFallback   bool
```

Comment:
Is this just so we can stage the rollout?  EG, 
1. make sure this is deployed to all ST cloud instances with `legacy_fallback=true`
2. Update the MT unified storage
3. update all ST cloud instances with `legacy_fallback=false`
4. remove the fallback
?

Reply:
Just to add more clarify: the `legacy_fallback` flag is only used on the Unified Storage side. On Grafana side, there's a feature flag: `appPlatformGrpcClientAuth`.

* `legacy_fallback`: when enabled, if the new authenticator fails, the legacy authenticator is tried, and the result of the legacy authenticator decides the outcome.
* `appPlatformGrpcClientAuth`: when enabled, the new client implementation is used on Grafana side, which means an ID token and/or Access token will be added to the request metadata, which is then verified on UniStore side. When this is disabled, the old client will be used, which will pass some basic identity information as metadata to allow the reconstruction of the identity on UniStor side (as a `StaticRequester`).

When all instances of Grafana and UniStor are running the latest code, we will be able to remove the flags, the code added to support these flags, and the legacy code (old client, old unistor authenticator).